PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -45,46 +45,70 @@ function WriteToScreenAndLog {
+        Write-Host "Connecting to EXO, please enter Global administrator credentials when prompted!" -ForegroundColor Yellow
         Connect-ExchangeOnline -ErrorAction Stop
-        $CurrentDescription= "Connecting to EXO V2"
+        $CurrentDescription= "Connecting to EXO"
         $CurrentStatus = "Success"
-        LogError -CurrentStatus $CurrentStatus -Function "Connecting to EXO V2" -CurrentDescription $CurrentDescription
-        Write-Host "Connected to EXO V2 successfully" -ForegroundColor Cyan
+        LogError -CurrentStatus $CurrentStatus -Function "Connecting to EXO" -CurrentDescription $CurrentDescription
+        Write-Host "Connected to EXO successfully" -ForegroundColor Cyan
     } catch {
         $ErrorEncountered=$Global:error[0].Exception
-        $CurrentDescription = "Connecting to EXO V2"
+        $CurrentDescription = "Connecting to EXO"
         $CurrentStatus = "Failure"
-        LogError -CurrentStatus $CurrentStatus -Function "Connecting to EXO V2" -CurrentDescription $CurrentDescription
+        LogError -CurrentStatus $CurrentStatus -Function "Connecting to EXO" -CurrentDescription $CurrentDescription
         Write-Host "Error encountered during executing the script!"-ForegroundColor Red
         Write-Host $ErrorEncountered -ForegroundColor Red
         Write-Host "`nOutput was exported in the following location: $ExportPath" -ForegroundColor Yellow
         Start-Sleep -Seconds 3
         break
     }
 }
+function ValidateDumpsterExistence {
+    param([PSCustomObject]$PublicFolder)
+    try {
+        $PublicFolderDumpster=Get-PublicFolder $PublicFolder.DumpsterEntryId -ErrorAction stop
+        $CurrentDescription = "Retrieving: $($PublicFolder.Identity) dumpster for diagnosing"
+        $CurrentStatus = "Success"
+        LogError -Function "Retrieve public folder $($PublicFolder.Identity) dumpster" -CurrentStatus $CurrentStatus -CurrentDescription $CurrentDescription
+        return $PublicFolderDumpster
+    } catch {
+        $Issue="Public folder $($PublicFolder.Identity) Dumpster is not existing!"
+        $Fix="FIX --> Please raise a support request for microsoft including the report & logs folder"
+        WriteToScreenAndLog -Issue $Issue -Fix $Fix
+        $ErrorEncountered=$Global:error[0].Exception
+        $CurrentDescription = "Retrieving: $($PublicFolder.Identity) dumpster for diagnosing"
+        $CurrentStatus = "Failure with error: "+$ErrorEncountered
+        LogError -Function "Retrieve public folder $($PublicFolder.Identity) dumpster" -CurrentStatus $CurrentStatus -CurrentDescription $CurrentDescription
+        if (!(Test-Path  "$ExportPath\logs_$ts")) {
+            mkdir "$ExportPath\logs_$ts" -Force | Out-Null
+        }
+        $PublicFolder | Export-Clixml -Path "$ExportPath\logs_$ts\PublicFolderInfo$($PublicFolder.Name).xml"
+        AskForFeedback
+        QuitEXOSession
```

Comment:
Why are we closing out the session in the middle of the script? We aren't going to end here because we handled the catch. It is going to continue to the next step after this function was called, unless you are expecting that to also then fail.

Reply:
Am quitting the script here as the main reason was found on the dumpster folder itself that requires Microsoft support so there's no need for the script to continue running.

Reply:
Except `QuitEXOSession` doesn't stop the script, it doesn't have an `exit` in it.

Reply:
Cause that will kill the PowerShell window as well that I'm using to preview blockers.

Reply:
If you do `exit` in a script, it `exit` the script not the PowerShell window. 

![image](https://github.com/microsoft/CSS-Exchange/assets/22776718/a53452ad-d073-41a2-88bb-d71e7a51c55d)

Reply:
Thanks for your example, added it.